src/goto-symex/shadow_memory_util.h

```diff
@@ -129,13 +129,17 @@ exprt compute_or_over_cells(
   const messaget &log,
   const bool is_union);
 
-// TODO: doxygen
+/// Aggregate the shadow memory value from the bytes of the multy-byte
+/// expression expr by applying the max function on the byte-values.
+/// \param expr the expression to extract the max from
+/// \param field_type the type of the shadow memory field to return
+/// \param ns the namespace to perform type-lookups into
+/// \return the aggregated max byte-sized value contained in expr
+/// Note that the expr type size must be known at compile time.
```

Comment:
In an invariant of `boolbv_widtht::operator()` that we use to get the size [here](https://github.com/diffblue/cbmc/pull/7935/files#diff-b8216daf2a64d1a9ccedb266114ed729520f219a57d77b3e0f372df029aa134fR524).

It would be nice to avoid the invariant by doing a check earlier, but unfortunately that's harder to do as it would involve a recursive check on the type structure.

If necessary it is possible to add it in, but I would defer it to a future PR